modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/DefaultPluginPatchesResolver.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.plugin;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.container.api.MuleFoldersUtil.getMuleHomeFolder;
+import static org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor.MULE_PLUGIN_CLASSIFIER;
+
+import static java.lang.String.format;
+import static java.nio.file.Files.walk;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+import static org.apache.commons.io.FilenameUtils.getExtension;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.module.artifact.activation.api.plugin.PluginPatchesResolver;
+import org.mule.tools.api.classloader.model.ArtifactCoordinates;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.aether.util.version.GenericVersionScheme;
+import org.eclipse.aether.version.InvalidVersionSpecificationException;
+import org.eclipse.aether.version.Version;
+import org.eclipse.aether.version.VersionConstraint;
+import org.slf4j.Logger;
+
+/**
+ * Default implementation of {@link PluginPatchesResolver}.
+ *
+ * @since 4.5
+ */
+public class DefaultPluginPatchesResolver implements PluginPatchesResolver {
```

Comment:
this should me moved down in the dependency tree with the migration of the descriptor generation usages, right?

Reply:
Exactly, I thought it would be better to move it during the migration. I'll add a TODO.